README.md

```diff
@@ -6,12 +6,120 @@
+| `Attribute` | `loadPrevious` | `<action-name>` | triggers associated user action when scroll reaches upper boundary |
+| `Attribute` | `loadNext` | `<action-name>` | triggers associated user action when scroll reaches lower boundary |
+
+## Examples
+### Basic list
+In order to follow the data down action up patten, `ember-frost-list` holds no state or information of current selected list items. When an item selection event emits, The user provided actions will be called and attrs will be provided to consumer as first param of the function with all useful information that describe the current select event
+
+In template
+```handlebars
+{{frost-list
+  componentPath
+  records=listItems
+  selections=selectedItems
+  onSelect=(action 'selected')
+}}
+```
+In controller
+```javascript
+componentPath: 'user-list-item',
+listItems: Ember.computed.alias('model'),
+selectedItems: [],
+actions: {
+  selected (attrs) {
+    if (attrs.isSelected) {
+      if (attrs.isShiftSelect) {
+        _.each(attrs.record, (record) => {
+          this.get('selectedItems').addObject(record)
+        })
+      } else {
+        if ((!attrs.isTargetSelectionIndicator && !attrs.isCtrlSelect)) this.set('selectedItems', [])
+        this.get('selectedItems').addObject(attrs.record)
+      }
+    } else {
+      this.get('selectedItems').removeObject(attrs.record)
+    }
+  }
+}
+```
+
+### Infinite scroll
+Implementing infinite scroll is straightforward. When scroll bar hits either upper boundary or lower boundary, loadPrevious/loadNext event will be emitted and consumer actions/functions will be called if provided.
+
+In template
+```handlebars
+{{frost-list
+  componentPath
+  records=listItems
+  selections=selectedItems
+  loadPrevious=(action 'loadPrevious')
+  loadNext=(action 'loadNext')
+  onSelect=(action 'selected')
+}}
+```
+In controller
+```javascript
+componentPath: 'user-list-item',
+listItems: Ember.computed.alias('model'),
+selectedItems: [],
+actions: {
+  _loadNext () {
+    // data retrieval/backend API call goes here
+  },
+
+  _loadPrevious () {
+    // data retrieval/backend API call goes here
+  },
+}
+```
+
+### integrated with sort and list item expansion/collapse control
+`ember-frost-list` uses `ember-block-slots` for multiple section yielding. A hash/object will be yielded back to the consumer which contains two contextual components named expansion and sort. The first is used for list item expand/collapse control and the second is obvious used for list sorting. Take a look at [ember-block-slots](https://github.com/ciena-blueplanet/ember-block-slots) and [ember-frost-sort](https://github.com/ciena-frost/ember-frost-sort) for detail usage.
+```handlebars
+{{#frost-list
+componentPath
+records=sortedItems
+selections=selectedItems
+onSelect=(action 'selectHandler')
+}}
+  {{#block-slot 'header' as |control|}}
+    {{control.sort sortableProperties=sortAttributes sortParams=querySortOrder onChange=(action 'sortHandler')}}
```

Comment:
Not your fault, but the sortParams is kind of an unclear property name.  We should raise an issue/asana task and fix that - something more like `sortOrder` - maybe we can change `sortableProperties` to just `properties` while we're at it.

Also a prime candidate for a mixin